app/routes.py

```diff
@@ -1,2 +1,114 @@
-from flask import Blueprint
+from flask import Blueprint, jsonify, abort, make_response, request
+from app import db
+from app.models.planet import Planet
+
+
+# planets = [
+#     Planet(1, "Arrakis", "A desert planet, source of Spice.", True),
+#     Planet(2, "Bela Tegeuse", "The fifth planet of Keuntsing.", False),
+#     Planet(3, "Tupile", "Sanctuary planet for defeated houses of the Imperium.", True),
+#     Planet(4, "Ix", "Supreme machine culture.", False)
+# ]
+
+planets_bp = Blueprint("Planets", __name__, url_prefix="/planets")
+
+# Helper Functions:
+
+def error_message(message, status_code):
+    abort(make_response(jsonify(dict(details=message)), status_code))
+
+def success_message(message, status_code=200):
+    return make_response(jsonify(dict(details=message)), status_code)
+
+def return_database_info(return_value):
+    return make_response(jsonify(return_value))
+
+def get_planet_record_by_id(id):
+    try:
+        id = int(id)
+    except ValueError:
+        error_message(f"Invalid id: {id}", 400)
+    planet = Planet.query.get(id)
+    if planet:
+        return planet
+    else:
+        error_message(f"Planet id: {id} not found", 404)
+
+def discover_planet_safely(data_dict):
+    try:
+        return Planet.create_from_dict(data_dict)
+    except ValueError as err:
+        error_message(f"Invalid key(s):{err}. Planet not added to database.", 400)
+    except KeyError as err:
+        error_message(f"Missing key(s): {err}.  Planet not added to database.", 400)
+
+
+def update_planet_safely(planet, data_dict):
+    try:
+        planet.update_self(data_dict)
+    except ValueError as err:
+        error_message(f"Invalid key(s): {err}. Planet not updated.", 400)
+    except KeyError as err:
+        error_message(f"Missing key(s): {err}. Planet not updated.", 400)
+
+
+
+
+# Route Functions:
+
+@planets_bp.route("", methods=["GET"])
+def get_all_planets():
+    moon_param = request.args.get("has_moons")
+    description_param = request.args.get("description")
+    if moon_param:
+        planets = Planet.query.filter_by(has_moons=moon_param)
```

Comment:
Does this work as expected for `moon_param`? Since this if-block is independent of the if/else below, when the request contains a `has_moons` query parameter and no `description` query parameter, it looks like the flow of execution would be that we hit line 63, enter the if and assign `planets`, then line 65 would be evaluated. Since there's no `description` query param we would enter the `else` and  the `planets` variable would be reassigned to `Planet.query.all()`.